### PR TITLE
Finished implementing outfit view and adding and saving outfits to it

### PR DESCRIPTION
**Pull request by aditiv1008**

### CODE REVIEW

File: app/src/main/java/com/example/pocketcloset/Fragments/OutfitsFragment.java [hunk 1]
```diff
@@ -51,15 +55,94 @@ public static OutfitsFragment newInstance(String param1, String param2) {
...
+            }
+        });
+
+        swipeContainer.setColorSchemeResources(android.R.color.holo_blue_bright,
+                android.R.color.holo_green_light,
+                android.R.color.holo_orange_light,
+                android.R.color.holo_red_light);
+
+
+        queryOutfits();
```
Comment (briantkelley) [inline app/src/main/java/com/example/pocketcloset/Fragments/OutfitsFragment.java:106]: nit: formatting

File: app/src/main/java/com/example/pocketcloset/MainActivity.java [hunk 2]
```diff
@@ -48,7 +50,10 @@ public boolean onNavigationItemSelected(@NonNull MenuItem item) {
                         fragment.setArguments(bundl);
                         break;
                     case R.id.action_outfits:
+                        Bundle bundle = new Bundle();
```
Comment (MrMannWood) [inline app/src/main/java/com/example/pocketcloset/MainActivity.java:53]: Ideally you'd want this to be a static function on the Fragment, so as to not expose the internals. At the very least, you want the key "userToFilterBy" to be a global constant.

File: app/src/main/java/com/example/pocketcloset/Fragments/OutfitsFragment.java [hunk 3]
```diff
@@ -51,15 +55,94 @@ public static OutfitsFragment newInstance(String param1, String param2) {
...
+        rvOutfits.setAdapter(adapter);
+        // set the layout manager on the recycler view
+        rvOutfits.setLayoutManager(new LinearLayoutManager(getContext()));
+
+
+        GridLayoutManager gridLayoutManager = new GridLayoutManager(
+                getContext(),
+                2 /* numberOfColumns */);
+
+        rvOutfits.setLayoutManager(gridLayoutManager);
```
Comment (MrMannWood) [inline app/src/main/java/com/example/pocketcloset/Fragments/OutfitsFragment.java:85]: You're overwriting the LinearLayoutManager that you've already set.

File: app/src/main/java/com/example/pocketcloset/Fragments/OutfitsFragment.java [hunk 4]
```diff
@@ -51,15 +55,94 @@ public static OutfitsFragment newInstance(String param1, String param2) {
...
+
+
+        swipeContainer = view.findViewById(R.id.swipeContainer);
+        //Setup refresh listener which triggers new data loading
+        swipeContainer.setOnRefreshListener(new SwipeRefreshLayout.OnRefreshListener() {
+            @Override
+            public void onRefresh() {
+                adapter.clear();
+                queryOutfits();
+                adapter.notifyDataSetChanged();
```
Comment (MrMannWood) [inline app/src/main/java/com/example/pocketcloset/Fragments/OutfitsFragment.java:95]: Generally speaking, I'd prefer to have the `notifyDataSetChanged` calls all be in the Adapter itself. This is normally done by calling it in the Adapter's method that gives it new data.

File: app/src/main/java/com/example/pocketcloset/Fragments/OutfitsFragment.java [hunk 5]
```diff
@@ -51,15 +55,94 @@ public static OutfitsFragment newInstance(String param1, String param2) {
...
+        query.include(Outfit.KEY_NECKWEAR);
+        query.include(Outfit.KEY_SHOES);
+        query.include(Outfit.KEY_OVERWEAR);
+        query.include(Outfit.KEY_HANDHELD);
+        query.include(Outfit.KEY_EARRINGS);
+        query.include(Outfit.KEY_HEADWEAR);
+        query.whereEqualTo(Outfit.KEY_USER, user);
+
+        // limit query to latest 20 items
+        query.setLimit(20);
```
Comment (MrMannWood) [inline app/src/main/java/com/example/pocketcloset/Fragments/OutfitsFragment.java:127]: why do we have this limit? Can the user ever set it to be more than 20?

File: app/src/main/java/com/example/pocketcloset/Fragments/OutfitsFragment.java [hunk 6]
```diff
@@ -51,15 +55,94 @@ public static OutfitsFragment newInstance(String param1, String param2) {
...
+        // order posts by creation date (newest first)
+        query.addDescendingOrder("createdAt");
+        // start an asynchronous call for posts
+        query.findInBackground(new FindCallback<Outfit>() {
+            @Override
+            public void done(List<Outfit> outfits, ParseException e) {
+                // check for errors
+                if (e != null) {
+                    Log.e("WardrobeFragment", "Issue with getting clothing", e);
```
Comment (MrMannWood) [inline app/src/main/java/com/example/pocketcloset/Fragments/OutfitsFragment.java:136]: you will eventually want to show an error message in these cases. I would recommend putting in a TODO with an easily-grepable string e.x. TODO add an error message

This way you can easily grep your code for the TODOs when you get around to adding the error messages, with a much lower chance of missing a case.

**Pull request merged**